### DISCUSSION TITLE
Adding workaround to directus install command to cover edge case

### DISCUSSION THREAD

**Discussion by eduardoPinto12** [directus/directus#10996] [Archive]
Hi guys, first thank you for all the hard work! 

I came across this PR https://github.com/directus/directus/issues/9911 which almost completely solves my problem except for one part. We use a _dockerfile_ to run directus on our production environment where we want to apply schema changes done locally and some migrations, but if we already have the db installed, the command `database install` will fail and won't allow us to run the `schema apply` and `database migrate` afterwards as it happens on `directus bootstrap`. Could we add the conditional to the install command to simply ignore the command if the db is already installed as it's different from a command error which is already caught with the catch?

Something along these lines:
```js
if ((await isInstalled()) === false) {
        try {
		await installSeeds(database);
		database.destroy();
		process.exit(0);
	} catch (err: any) {
		logger.error(err);
		database.destroy();
		process.exit(1);
	}
}
```

Thanks

**Comment by rijkvanzanten** [comment]
That makes total sense to me. When the database is already installed, the installer shouldn't try again. The only thing I'm wondering about is whether or not it should throw an error 🤔 

If you try installing, and the database is already (successfully) installed, should it consider that a successful run?

**Reply by eduardoPinto12** [reply]
I get your point, as it isn't actually running/completing the command :sweat_smile: but in my opinion it should, because the outcome was also reached (previously of course) but nonetheless reached. For the directus bootstrap we consider that instruction to be a successful one on the same scenario or just an ignored one? The DoD/definition of successful of this step is indeed tricky for this scenario :smile:

**Reply by rijkvanzanten** [reply]
> For the directus bootstrap we consider that instruction to be a successful one on the same scenario or just an ignored one?

Yeah! The idea is that bootstrap just makes sure your database is up to date, so even if it doesn't have to actually run anything, the outcome is successful, as the database is up to date 🤔 

The same could be set for install!

**Reply by eduardoPinto12** [reply]
Regarding throwing an error I was thinking about mimicking the exact same behavior of bootstrap and simply inform the user that the database was previously installed so there wasn't a need to run the command

**Reply by rijkvanzanten** [reply]
Yeah, that makes sense!

**Reply by eduardoPinto12** [reply]
I also saw this [comment](https://github.com/directus/directus/pull/7939#issuecomment-916417750) of yours which could also be a good solution, not sure if it was implemented but I can't find it on `directus bootstrap` command, that way we wouldn't need `database install` and could just use `directus bootstrap` instead. Whichever you think is best :+1:

**Reply by rijkvanzanten** [reply]
Ah, yeah that doesn't exist as of right now. Happy to hear your thoughts on what you think is best 🙂 

@keesvanbemmel and I have been discussing this in more detail as of late as well; super tricky subject to get _just right_

**Reply by eduardoPinto12** [reply]
@rijkvanzanten I can imagine :sweat_smile:  but do you know if there's any plan for this soon? We really needed it for our production environment and was wondering if we could open a PR with changes to database install command for now or if there's any plan for bootstrap or database install in the near future :pray:

**Reply by rijkvanzanten** [reply]
I can't promise anything around timing for new features / improvements that aren't [sponsored](https://docs.directus.io/getting-started/support.html#sponsored-work) unfortunately. That being said, do you have a more in-depth plan on how this should work ideally for your use case? If so, it be great if you could put together a more in-depth [RFC](https://github.com/directus/rfcs) so we can discuss final strategy, and figure out how to implement things most efficiently 🙂

**Reply by eduardoPinto12** [reply]
Sorry got into PTO in the meantime :sweat_smile: My company could not wait unfortunately :cry: So we decided to migrate all changes including UI through the migrations option and using bootstrap, thanks anyway @rijkvanzanten :+1: I'll see if I can get some of my personal time to maybe think about this and contribute for other users
